nbgrader/converters/base.py

```diff
@@ -293,6 +294,9 @@ def _handle_failure(gd):
                 for notebook_filename in self.notebooks:
                     self.convert_single_notebook(notebook_filename)
 
+                # create the archive
+                self.make_archive();
```

Comment:
Also, I think this isn't quite the right spot to have it actually create the archive, because this will cause nbgrader to create a zip archive for all steps of the process--not just `nbgrader assign`. I think what you want to do is probably to add a new "dummy" function to the base class (called something like `after_assignment_converted()`, or something alone those lines) that by default does nothing. Then, in the `Assign` class, override that function to call the archiving machinery (and then move `make_archive` and `recursively_archive` into the `Assign` class as well).